src/components/Reports/ReportsDailyVisits/ReportsDailyVisits.js

```diff
@@ -0,0 +1,79 @@
+import React, { useEffect, useState } from 'react';
+import { InjectAppServices } from '../../../services/pure-di';
+import { FormattedMessage } from 'react-intl';
+import Loading from '../../Loading/Loading';
+import C3Chart from '../../shared/C3Chart/C3Chart';
+
+let chartConfig = {
+  data: {
+    json: null,
+    type: 'spline',
+    keys: {
+      x: 'from',
+      value: ['quantity'],
+    },
+  },
+  legend: {
+    show: false,
+  },
+  axis: {
+    x: {
+      type: 'timeseries',
+      tick: {
+        format: '%Y-%m-%d',
+      },
+    },
+  },
+};
+
+const ReportsDailyVisits = ({ domainName, dateFrom, dependencies: { datahubClient } }) => {
+  const [state, setState] = useState({ loading: true });
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const dailyVisitsData = await datahubClient.getDailyVisitsByPeriod({
+        domainName: domainName,
+        dateFrom: dateFrom,
+      });
+      if (!dailyVisitsData.success || !dailyVisitsData.value) {
+        setState({ loading: false });
+      } else {
+        const data = dailyVisitsData.value.map((item) => {
+          let date = new Date(item.from);
+          date = date.getFullYear() + '-' + (date.getMonth() + 1) + '-' + date.getDate();
+          return {
+            quantity: item.quantity,
+            from: date,
+          };
+        });
+        chartConfig.data.json = data;
```

Comment:
What do you think about creating a new object in place of modifying `chartConfig`?

```suggestion
        const chartData = { ...chartConfig, data: { ...chartConfig, json: data } };
```

(I am not sure if this _deconstructed_ syntax is totally right)

Reply:
Fixed